### PR TITLE
Updated darc documentation

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

File: Documentation/Darc.md [hunk 1]
```diff
@@ -1,447 +1,1691 @@
...
+
+  delete-default-channel       Remove a default channel association.
+
+...
+```
+
+When executing most operations, the client needs to make some remote queries.
+These remote queries require authentication in most circumstances. There are 3
+PATs that may be used:
+- A GitHub PAT for downloading files off github (e.g. eng/Version.Details.xml or
```
Comment (garath) [inline Documentation/Darc.md]:
```suggestion
- A GitHub PAT for downloading files off GitHub (e.g. eng/Version.Details.xml or
```

File: Documentation/Darc.md [hunk 2]
```diff
@@ -1,447 +1,1691 @@
...
+information about each one:
+- Name
+- Source sha
+- Source repository
+- Is the dependency version pinned (can be it automatically updated?)
+- Dependency type (toolset or product)
 
-### Arcade updating its Arcade dependency using Maestro++ and Darc
+The [`darc add`](#add) command adds a new dependency.  It takes a number of
+parameters, though only `--name` and `--type` are initially required.  It is
```
Comment (garath) [inline Documentation/Darc.md]: What makes the repo parameter highly recommended?

Consider rewording like "The `--repo` parameter is optional but highly recommended to improve traceability."

File: Documentation/Darc.md [hunk 3]
```diff
@@ -1,447 +1,1691 @@
...
+dependency information.  The information can also be filled in after commiting by Maestro++
+using dependency flow, but passing your CI without a filled-in version number
+may be difficult. See (Updating dependencies in your local repository).
+
+#### Toolset vs. Product Dependencies
+
+There are two types of dependencies: 'Product' and 'Toolset'. Choosing between
+them involves answering the question:
+
+"Does my repository repackage the input dependency's binaries, or information
```
Comment (garath) [inline Documentation/Darc.md]: Which answer goes with which option?

File: Documentation/Darc.md [hunk 4]
```diff
@@ -1,447 +1,1691 @@
...
+For example, a build of dotnet/corefx might be applied to the ".NET Core 3 Dev"
+channel. dotnet/core-setup maps new outputs of corefx on the ".NET Core 3 Dev"
+channel onto its master branch.
+
+A subscription has a few parts:
+- Mapping of source repo + source channel => target repo target branch
+- An update rate (e.g. every day, every build, not at all)
+- Whether a subscription is batchable or not. If batchable, all batchable
+  subscriptions targeting the same branch/repo combination will share a PR.
+  *Note: Batchable subscriptions are currently unsupported in darc*
```
Comment (garath) [inline Documentation/Darc.md]: Describing features that do not yet exist just adds maintenance liability. Better to not mention it at all until after it is released.

Comment (mmitche) [reply Documentation/Darc.md]: JC is working on it right now I think.

File: Documentation/Darc.md [hunk 5]
```diff
@@ -1,447 +1,1691 @@
...
+
+Whenever you want regular dependency flow between two repositories. For example,
+every day (when there is a new source build) or on every new build.
+
+#### What input channel should be used?
+
+There are generally two channels for day to day use:
+- '.NET Core 3 Dev' - Day to day builds of .NET Core 3 repositories are placed on
+  this channel.
+- '.NET Tools - Latest' - LKG Arcade builds are placed on this channel.
```
Comment (garath) [inline Documentation/Darc.md]:
```suggestion
- '.NET Tools - Latest' - Arcade releases are placed on this channel.
```

File: Documentation/Darc.md [hunk 6]
```diff
@@ -1,447 +1,1691 @@
...
+
+- `-p, --password` - Build Asset Registry password.  You can obtain this
+  password by going to https://maestro-prod.westus2.cloudapp.azure.com/, logging
+  in using the link in the top right, then generating a token using the menu in
+  the top right.  This setting overrides whatever BAR password was provided when
+  doing `darc authenticate`.
+- `--github-pat` - Person access token used to authenticate GitHub. This is a GitHub PAT used
+  to avoid rate limiting when accessing github to download arcade script files
+  or version files. You only need a GitHub PAT with **no** authorization scopes
+  checked. This setting overrides whatever BAR password was provided when
```
Comment (garath) [inline Documentation/Darc.md]:
```suggestion
  checked. This setting overrides whatever GitHub PAT was provided when
```

File: Documentation/Darc.md [hunk 7]
```diff
@@ -1,447 +1,1691 @@
...
+  doing `darc authenticate`.
+- `--azdev-pat` - Personal access token used to authenticate to Azure DevOps.
+  This token should have Code Read permissions. This setting overrides whatever BAR password was provided when
```
Comment (garath) [inline Documentation/Darc.md]:
```suggestion
  This token should have Code Read permissions. This setting overrides whatever Azure DevOps PAT was provided when
```

File: Documentation/Darc.md [hunk 8]
```diff
@@ -1,447 +1,1691 @@
...
+  doing `darc authenticate`.
+- `--bar-uri` - URI of the build asset registry service to use.  Typically left
+  as its default (https://maestro-prod.westus2.cloudapp.azure.com) This setting overrides whatever BAR password was provided when
```
Comment (garath) [inline Documentation/Darc.md]: Text on "BAR password" seems like a copy-paste error?

File: Documentation/Darc.md [hunk 9]
```diff
@@ -1,447 +1,1691 @@
...
+
+## Index
+- [Scenarios](#scenarios)
+  - [Setting up your darc client](#setting-up-your-darc-client)
+  - [Adding dependencies to a repository](#adding-dependencies-to-a-repository)
+  - [Updating dependencies in your local repository](#updating-dependencies-in-your-local-repository)
+  - [Removing dependencies from a repository](#removing-dependencies-from-a-repository)
+  - [Changing a dependency's type](#changing-a-dependencys-type)
+  - ['Pinning' dependencies so they do not
+    update](#pinning-dependencies-so-they-do-not-update)
```
Comment (jcagme) [inline Documentation/Darc.md]: Nit: This can be in the same line

File: Documentation/Darc.md [hunk 10]
```diff
@@ -1,447 +1,1691 @@
...
+  - [add-subscription](#add-subscription) - Add a new subscription.
+  - [authenticate](#authenticate) - Stores the VSTS and GitHub tokens required for remote operations.
+  - [delete-channel](#delete-channel) - Deletes an existing channel.
+  - [delete-default-channel](#delete-default-channel) - Remove a default channel association.
+  - [delete-subscription](#delete-subscription) - Remove a subscription.
+  - [gather-drop](#gather-drop) - Gather a drop of the outputs for a build
+  - [get-channels](#get-channels) - Get a list of channels.
+  - [get-default-channels](#get-default-channels) - Gets a list of repo+branch combinations and their associated default channels for builds.
+  - [get-dependencies](#get-dependencies) - Get local dependencies.
+  - [get-dependency-graph](#get-dependency-graph) - Get local dependencies.
```
Comment (jcagme) [inline Documentation/Darc.md]: >Get local dependencies. [](start = 52, length = 23)

Update (copy/paste from above)

File: Documentation/Darc.md [hunk 11]
```diff
@@ -1,447 +1,1691 @@
...
+  - [Common Parameters](#common-parameters)
+  - [add-channel](#add-channel) - Creates a new channel.
+  - [add](#add) - Add a new dependency to Version.Details.xml.
+  - [add-default-channel](#add-default-channel) - Add a channel that a build of a branch+repository is automatically applied to.
+  - [add-subscription](#add-subscription) - Add a new subscription.
+  - [authenticate](#authenticate) - Stores the VSTS and GitHub tokens required for remote operations.
+  - [delete-channel](#delete-channel) - Deletes an existing channel.
+  - [delete-default-channel](#delete-default-channel) - Remove a default channel association.
+  - [delete-subscription](#delete-subscription) - Remove a subscription.
+  - [gather-drop](#gather-drop) - Gather a drop of the outputs for a build
```
Comment (jcagme) [inline Documentation/Darc.md]: >d [](start = 73, length = 1)

Missing '.'

File: Documentation/Darc.md [hunk 12]
```diff
@@ -1,447 +1,1691 @@
...
+
+...
+```
+
+When executing most operations, the client needs to make some remote queries.
+These remote queries require authentication in most circumstances. There are 3
+PATs that may be used:
+- A GitHub PAT for downloading files off github (e.g. eng/Version.Details.xml or
+  arcade script files. 
+- An Azure DevOps pat for downloading files off of Azure DevOps. (e.g.
```
Comment (jcagme) [inline Documentation/Darc.md]: >pat [](start = 18, length = 3)

pat -> PAT

File: Documentation/Darc.md [hunk 13]
```diff
@@ -1,447 +1,1691 @@
...
+repository and opening PRs in other repositories when those repositories' input
+dependencies in eng/Version.Details.xml match the outputs of the new builds.
+
+These updates are selective and based on 'subscriptions'. A subscription describes an update
+operation for a specific repository+branch combination, mapping outputs of a
+repository that have beeen applied to a channel (virtual branch) onto matching
+inputs of the target repository+branch.
+
+For example, a build of dotnet/corefx might be applied to the ".NET Core 3 Dev"
+channel. dotnet/core-setup maps new outputs of corefx on the ".NET Core 3 Dev"
```
Comment (jcagme) [inline Documentation/Darc.md]: >corefx [](start = 47, length = 6)

dotnet/corefx

File: Documentation/Darc.md [hunk 14]
```diff
@@ -1,447 +1,1691 @@
...
+channel onto its master branch.
+
+A subscription has a few parts:
+- Mapping of source repo + source channel => target repo target branch
+- An update rate (e.g. every day, every build, not at all)
+- Whether a subscription is batchable or not. If batchable, all batchable
+  subscriptions targeting the same branch/repo combination will share a PR.
```
Comment (jcagme) [inline Documentation/Darc.md]: >branch/repo [](start = 35, length = 11)

repo/branch?

File: Documentation/Darc.md (hunk 4 above)
Comment (jcagme) [reply Documentation/Darc.md]: Will come after updates which is mostly done :)

---
In reply to: [253236648](https://github.com/dotnet/arcade/pull/1951#discussion_r253236648) [](ancestors = 253236648)

File: Documentation/Darc.md [hunk 15]
```diff
@@ -1,447 +1,1691 @@
...
+  *Note: Batchable subscriptions are currently unsupported in darc*
+- A set of auto merge policies, if the subscription is not batchable.  If batchable,
+  merge policies are set on a repository level rather than a per-subscription
+  level, as they end up shared between several subscriptions. *Note: repository
+  merge policies are currently unsupported in darc*
+
+For additional information and samples, see [add-subscription](#add-subscription)
+
+#### When should a subscription be used to update my repository's inputs
```
Comment (jcagme) [inline Documentation/Darc.md]: >s [](start = 71, length = 1)

Add '?' at the end

File: Documentation/Darc.md [hunk 16]
```diff
@@ -1,447 +1,1691 @@
...
+Gathering a drop of the product or subpiece part of the product is a typical
+activity done by developers, release automation, etc.  The darc tool can do this
+using the Build Asset Registry information alongside the dependency graph
+information.
+
+A build drop is a gathering of all the outputs of a specific build that were
+reported to the Build Asset Registry into a local location. Optionally, this
+build drop may also include any builds that were inputs to this build, based on
+the dependency information in Version.Details.xml. Builds produce various
+outputs that go to various locations. For example, they may produce nuget
```
Comment (jcagme) [inline Documentation/Darc.md]: >nuget [](start = 68, length = 5)

nuget -> NuGet

File: Documentation/Darc.md [hunk 17]
```diff
@@ -1,447 +1,1691 @@
...
+packges, zips, msis, etc. These may be available in the build artifacts, or may
```
Comment (jcagme) [inline Documentation/Darc.md]: >msis [](start = 15, length = 4)

msis -> MSIs

File: Documentation/Darc.md [hunk 18]
```diff
@@ -1,447 +1,1691 @@
...
+dependencies are those that generally serve as inter-repo transport.  Some
+examples:
+- Packages shipping to nuget.org - Shipping
+- Coreclr transport packages - Nonshipping, since they get repackaged by
+  core-setup
+- Symbols - Shipping, since they go to symbol servers
+
+#### What can I gather a drop for?
+
+You can gather a either an isolated or full (includes dependencies) drop for any
```
Comment (jcagme) [inline Documentation/Darc.md]: > a  [](start = 14, length = 3)

I think this 'a' is not needed

File: Documentation/Darc.md [hunk 19]
```diff
@@ -1,447 +1,1691 @@
...
+
+`gather-drop` must be able to find a root build. This root build today is
+identified by either a repo+sha combo or a direct BAR build ID. For most
+repositories that build on every commit, providing the repo uri and latest sha
+will generate a drop:
+
+```
+PS C:\enlistments\core-sdk> git rev-parse HEAD
+465a336c7a5ca3af2f6cf5172ddc0ebde620803b
+PS C:\enlistments\core-sdk> darc gather-drop --repo https://github.com/dotnet/core-sdk --commit 465a336c7a5ca3af2f6cf5172ddc0ebde620803b --output-dir C:\scratch\drop
```
Comment (jcagme) [inline Documentation/Darc.md]: >PS C:\enlistments\core-sdk>  [](start = 0, length = 28)

Split this line in two so the length matches the rest of the doc

File: Documentation/Darc.md [hunk 20]
```diff
@@ -1,447 +1,1691 @@
...
+### **`Common parameters`**
+
+There are a few common parameters available on every command:
+
+- `-p, --password` - Build Asset Registry password.  You can obtain this
+  password by going to https://maestro-prod.westus2.cloudapp.azure.com/, logging
+  in using the link in the top right, then generating a token using the menu in
+  the top right.  This setting overrides whatever BAR password was provided when
+  doing `darc authenticate`.
+- `--github-pat` - Person access token used to authenticate GitHub. This is a GitHub PAT used
```
Comment (jcagme) [inline Documentation/Darc.md]: >Person [](start = 19, length = 6)

Person -> Personal

File: Documentation/Darc.md [hunk 21]
```diff
@@ -1,447 +1,1691 @@
...
-            <Sha>132412342341234234</Sha>
-        </Dependency>
-    </ToolsetDependencies>
+**See also**:
+- [delete-channel](#delete-channel)
+- [get-channels](#get-channels)
+
+### **`add`**
+
+Add a new tracked dependency to the Version.Detail.xml file in your local repo.
```
Comment (jcagme) [inline Documentation/Darc.md]: >Add a new tracked dependency to the Version.Detail.xml [](start = 0, length = 54)

'add' also updates Versions.props and global.json depending on the added dependency. Might be worth mentioning that

File: Documentation/Darc.md [hunk 22]
```diff
@@ -1,447 +1,1691 @@
...
+      <Sha>14d1133b6074b463784a7adbbf385df0462f4010</Sha>
+    </Dependency>
+  </ToolsetDependencies>
+</Dependencies>
+```
+
+Running add
+
+```
+PS D:\enlistments\arcade> darc add --name "Microsoft.NETCore.App" --type "product" --version 1 --commit 2 --repo https://github.com/dotnet/core-setup
```
Comment (jcagme) [inline Documentation/Darc.md]: >PS D:\enlistmen [](start = 0, length = 15)

Split in two

File: Documentation/Darc.md [hunk 23]
```diff
@@ -1,447 +1,1691 @@
...
+- [update-dependencies](#update-dependencies)
+- [Adding dependencies to a repository](#adding-dependencies-to-a-repository)
+
+### **`add-default-channel`**
+
+Adds a new default channel mapping.  A default channel maps each new build of a specific
+branch of a repository onto a specific channel. While builds can be selectively
+and manually applied to channels, this is generally inconvenient for day to day development
+in most cases.  In general, until release shutdown, each build of a branch
+should always be applied to its "normal" channel.,
```
Comment (jcagme) [inline Documentation/Darc.md]: >, [](start = 49, length = 1)

Remove ','

File: Documentation/Darc.md [hunk 24]
```diff
@@ -1,447 +1,1691 @@
...
+operation for a specific repository+branch combination, mapping outputs of a
+repository that have beeen applied to a channel (virtual branch) onto matching
+inputs of the target repository+branch.
+
+For example, a build of dotnet/corefx might be applied to the ".NET Core 3 Dev"
+channel. dotnet/core-setup maps new outputs of corefx on the ".NET Core 3 Dev"
+channel onto its master branch.
+
+A subscription has a few parts:
+- Mapping of source repo + source channel => target repo target branch
```
Comment (jcagme) [inline Documentation/Darc.md]: >target repo target branch [](start = 45, length = 25)

target repo + target branch

File: Documentation/Darc.md [hunk 25]
```diff
@@ -1,447 +1,1691 @@
...
+- An update rate (e.g. every day, every build, not at all)
+- Whether a subscription is batchable or not. If batchable, all batchable
+  subscriptions targeting the same branch/repo combination will share a PR.
+  *Note: Batchable subscriptions are currently unsupported in darc*
+- A set of auto merge policies, if the subscription is not batchable.  If batchable,
+  merge policies are set on a repository level rather than a per-subscription
+  level, as they end up shared between several subscriptions. *Note: repository
+  merge policies are currently unsupported in darc*
+  
+`add-subscription` has two modes of operation
```
Comment (jcagme) [inline Documentation/Darc.md]: >operation [](start = 36, length = 9)

operation -> operation:

File: Documentation/Darc.md [hunk 26]
```diff
@@ -1,447 +1,1691 @@
...
+- `--target-repo` - **(Required if -q is passed)** Target repository for the subscription.  Builds of
+  `--source-repo` that appear on the specified `--channel` will have matching
+  outputs applied to the inputs (specified in eng/Version.Details.xml) of this
+  repo's `--target-branch`
+- `--target-branch` - **(Required if -q is passed)** Target branch for the subscription. Builds of
+  `--source-repo` that appear on the specified `--channel` will have matching
+  outputs applied to the inputs (specified in eng/Version.Details.xml) on this
+  branch of `--target-repo`.
+- `--update-frequency` - **(Required if -q is passed)** Frequency of updates. Valid values are: 'none',
+  'everyDay', or 'everyBuild'.  Every day is applied at 5am.  Subscriptions with
```
Comment (jcagme) [inline Documentation/Darc.md]: >Every day [](start = 32, length = 9)

Every day -> 'everyDay'

File: Documentation/Darc.md [hunk 27]
```diff
@@ -1,447 +1,1691 @@
...
+  'none' frequency can still be triggered using [trigger-subscriptions](#trigger-subscriptions)
+- `--all-checks-passed` - Merge policy. A PR is automatically merged by Maestro++ if there is at least one
+  check and all are passed. Optionally provide a comma separated list of
+  ignored check with --ignore-checks.
+- `--ignore-checks` - Merge policy. A For use with --all-checks-passed. A set of checks that are
```
Comment (jcagme) [inline Documentation/Darc.md]: >A For [](start = 36, length = 5)

A For -> For(?)

File: Documentation/Darc.md [hunk 28]
```diff
@@ -1,447 +1,1691 @@
...
+  ignored. Typically, in github repos the "WIP" and "license/cla" checks are ignored.
+- `--no-extra-commits`- Merge policy. A PR is automatically merged if no non-bot commits exist in the PR.
+- `--require-checks` - Merge policy. A PR is automatically merged if the specified checks are passed. Provide a comma separate list of required checks.
```
Comment (jcagme) [inline Documentation/Darc.md]: >`--require-checks [](start = 2, length = 17)

Split

File: Documentation/Darc.md [hunk 29]
```diff
@@ -1,447 +1,1691 @@
...
+- `-q, --quiet` - Non-interactive mode (requires all elements to be passed on the command line).
+
+**Sample**:
+```
+PS D:\enlistments\arcade-services> darc add-subscription --channel ".NET Tools - Latest" --source-repo https://github.com/dotnet/arcade --target-repo https://dev.azure.com/dnceng/internal/_git/dotnet-optimization --target-branch master --update-frequency everyDay --all-checks-passed -q
```
Comment (jcagme) [inline Documentation/Darc.md]: >PS D:\enlistments\ [](start = 0, length = 18)

Split

File: Documentation/Darc.md [hunk 30]
```diff
@@ -1,447 +1,1691 @@
...
+build_asset_registry_base_uri=https://maestro-prod.westus2.cloudapp.azure.com/
+
+# Storing the required settings...
+# Set elements above depending on what you need
+
+```
+
+### **`delete-channel`**
+
+Delete a channel. This channel must be in use by any subscriptions.
```
Comment (jcagme) [inline Documentation/Darc.md]: >This channel must be in use by any subscriptions. [](start = 18, length = 49)

So, if I create a channel that I realize after that I won't use so it is not used by any subscription, I won't be able to delete it?

File: Documentation/Darc.md [hunk 31]
```diff
@@ -1,447 +1,1691 @@
...
+```
+
+**See also**:
+- [add-channel](#add-channel)
+- [get-channels](#get-channels)
+
+### **`delete-default-channel`**
+
+Deletes a default channel mapping. Deleting will not affect any existing builds,
+but new builds of the specified repos will will not be applied to the target
```
Comment (jcagme) [inline Documentation/Darc.md]: >will [](start = 43, length = 4)

Dup

File: Documentation/Darc.md [hunk 32]
```diff
@@ -1,447 +1,1691 @@
...
+[get-default-channels](#get-default-channels)
+
+- `--channel` - **(Required)** Name of channel that builds of 'repository' and 'branch' should not apply to.
+- `--branch` - **(Required)** Repository that should have its default association removed.
+- `--repo` - **(Required)** Branch that should have its default association
+  removed.
+
+**Sample**
+```
+PS D:\enlistments\arcade> darc delete-default-channel --channel ".Net Core 3 Dev" --branch refs/heads/master --repo https://github.com/dotnet/arcade
```
Comment (jcagme) [inline Documentation/Darc.md]: >PS D:\enlis [](start = 0, length = 11)

Break into more lines

File: Documentation/Darc.md (hunk 17 above)
Comment (jcagme) [reply Documentation/Darc.md]: Both of the above also apply to line 1027

---
In reply to: [253530424](https://github.com/dotnet/arcade/pull/1951#discussion_r253530424) [](ancestors = 253530424)

File: Documentation/Darc.md [hunk 33]
```diff
@@ -1,447 +1,1691 @@
...
+- `-c, --commit` - Branch, commit or tag to look up and gather a build drop for.
+  For information on locating the
+  "root build", see [Gathering a build drop](#gathering-a-build-drop)
+- `-o, --output-dir` - **(Required)** Output directory to place build drop.
+- `-f, --full` - Gather the full drop (build and all input builds).
+- `-s, --separated` - Separate out each source repo in the drop into separate directories.
+- `--continue-on-error` - Continue on error rather than halting.  Allows for
+  gathering drops in cases where some outputs might not be able to be
+  downloaded.
+- `--non-shipping`` - (Default: true) Include non-shipping assets.
```
Comment (jcagme) [inline Documentation/Darc.md]: >`` [](start = 17, length = 2)

Remove one '`'

File: Documentation/Darc.md [hunk 34]
```diff
@@ -1,447 +1,1691 @@
...
-	{
-		name: "Microsoft.DotNet.Build.Tasks.Feed",
-		version: "1.0.0-prerelease-02201-02",
-		repo-uri: "https://github.com/dotnet/buildtools",
-		sha: "23498123740982349182340981234",
-		type: "toolset"
-	}
-]
+PS C:\enlistments\core-sdk> dotnet C:\enlistments\arcade-services\artifacts\bin\Microsoft.DotNet.Darc\Debug\netcoreapp2.1\Microso
+ft.DotNet.Darc.dll gather-drop --output-dir C:\scratch\core-sdk-drop --commit 465a336c7a5ca3af2f6cf5172ddc0ebde620803b --repo https://github.com/dotnet/core-sdk
```
Comment (jcagme) [inline Documentation/Darc.md]: >ft.D [](start = 0, length = 4)

Break into lines (here and elsewhere applicable)

File: Documentation/Darc.md (hunk 2 above)
Comment (mmitche) [reply Documentation/Darc.md]: Without --repo you can't immediately run --update-dependencies.  I'll rework this

File: Documentation/Darc.md [hunk 35]
```diff
@@ -1,447 +1,1691 @@
...
+Looking up Microsoft.Build@16.0.0-preview.386 in Build Asset Registry...
+Looking up build 2010 in Build Asset Registry...
+Finding build for Microsoft.NETCore.Compilers@3.0.0-beta3-19067-14...
+Looking up Microsoft.NETCore.Compilers@3.0.0-beta3-19067-14 in Build Asset Registry...
+Looking up build 1598 in Build Asset Registry...
+Finding build for Microsoft.NET.Sdk.Razor@3.0.0-preview-19079-02...
+Looking up Microsoft.NET.Sdk.Razor@3.0.0-preview-19079-02 in Build Asset Registry...
+Looking up build 2015 in Build Asset Registry...
+Finding build for Microsoft.NET.Sdk.WindowsDesktop@3.0.0-preview-27329-5...
+Lo
```
Comment (jcagme) [inline Documentation/Darc.md]: >displayed.  [](start = 0, length = 11)

Remove and add new break.

Comment (jcagme) [inline Documentation/Darc.md]: >s [](start = 62, length = 1)

Add '.'

File: Documentation/Darc.md (hunk 30 above)
Comment (mmitche) [reply Documentation/Darc.md]: must not be.

File: Documentation/Darc.md (hunk 35 above)
Comment (jcagme) [inline Documentation/Darc.md]: >/ [](start = 69, length = 1)

/ -> .

File: Documentation/Darc.md [hunk 36]
```diff
@@ -1,447 +1,1704 @@
...
+- [trigger-subscriptions](#trigger-subscriptions)
+- [get-channels](#get-channels)
+
+### **`authenticate`**
+
+Set up your darc client so that the PAT or password inputs do not need to be
+passed on each command invocation.  This command opens up an editor form with
+various password settings. These values are overridden by the `--password`,
+`--bar-uri`, `--azdev-pat` and `--github-pat` settings common to all commands.
+
```
Comment (garath) [inline Documentation/Darc.md]: I suggest highlighting the required scopes for the PATs. They are mentioned in the example below, but up here would be more noticeable.

File: Documentation/Darc.md [hunk 37]
```diff
@@ -1,447 +1,1704 @@
...
+  - [Viewing the dependency graph](#viewing-the-dependency-graph)
+  - [Gathering a build drop](#gathering-a-build-drop)
+  
+- [Command Reference](#command-reference)
+  - [Common Parameters](#common-parameters)
+  - [add-channel](#add-channel) - Creates a new channel.
+  - [add](#add) - Add a new dependency to Version.Details.xml.
+  - [add-default-channel](#add-default-channel) - Add a channel that a build of a branch+repository is automatically applied to.
+  - [add-subscription](#add-subscription) - Add a new subscription.
+  - [authenticate](#authenticate) - Stores the VSTS and GitHub tokens required for remote operations.
```
Comment (garath) [inline Documentation/Darc.md]:
```suggestion
  - [authenticate](#authenticate) - Stores the Azure DevOps and GitHub tokens required for remote operations.
```

**Pull request merged**